Software/Development/SemesterProject/App/openLoopControlManager.c

```diff
@@ -0,0 +1,80 @@
+const float vfConst = 0.0185185; //1/54Hz
+
+
+/*
+ * mapFloat maps the value from one reference to another.
+ * Assuming only floats and minimum value of 0 for x and y.
+ */
+float mapFloat(float x, float xMax, float yMax)
+{
+    return x * yMax / xMax;
+}
+
+int16 mapInt(int16 x, int16 xMax, int16 yMax)
+{
+    return x * yMax / xMax;
+}
+
+
+
+void setConstantDutyFromPot(void)
+{
+    setAllDuties(mapFloat(1.2, 3.3, readMaxDuty())); //mapFloat(readADC(), readMaxADC(), readMaxDuty()));
+}
+
+
+//
+//This function is called at switching frequency so the duty needs to get updated in every iteration.
+//
+void setFrequencyFromPot(void)
+{
+    float statorFreq;
+    float stepsPerCycle;
+    float dutyA, dutyB, dutyC;
+    int i = 0;
+    float vfFactor;
+
+    statorFreq = mapFloat(getTorqueReferenceSliderMeasurement(), 3.3, MAX_STATOR_FREQ); //substitute the 3.3 for getMaximumReferenceValue() in analogAcquisitionManager.c
+
+    vfFactor = statorFreq * vfConst;
+
+    stepsPerCycle = statorFreq * SINE_LUT_LENGTH / SW_FREQ;
+
+    currentCyclePos[0] += stepsPerCycle;
+    //Update current cycle position
+    for (i = 0; i < 3; i++)
+    {
+        currentCyclePos[i] = currentCyclePos[0] + i * SINE_LUT_LENGTH/3;
+        if (currentCyclePos[i] >= SINE_LUT_LENGTH) {currentCyclePos[i] -= SINE_LUT_LENGTH;}
+    }
+
+
+    dutyA = vfFactor * sineLut[(int16) currentCyclePos[0]]/10;
```

Comment:
if the casting is done because of the previous division, I think it's better to do the casting in the division line: It then beecomes evident why is done.

Reply:
The casting here is done as a way to approximate the float to an int16 to have access to the LUT vector.

Reply:
My point is that the casting should be done at the moment that the float is generated. Do you think that will create glitches in the output waveform?

Reply:
As we talked in the grouproom, the casting is done after the currentCyclePos is set to have better precision in the vector value.